File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailProvider.java
package fr.free.nrw.commons.media;

import fr.free.nrw.commons.Media;

public interface MediaDetailProvider {

    Media getMediaAtPosition(int i);

    int getTotalMediaCount();

    Integer getContributionStateAt(int position);

    // Reload media detail fragment once media is nominated
    void refreshNominatedMedia(int index);
}

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailProvider.kt
package fr.free.nrw.commons.media

import fr.free.nrw.commons.Media

interface MediaDetailProvider {
    fun getMediaAtPosition(i: Int): Media?

    fun getTotalMediaCount(): Int

    fun getContributionStateAt(position: Int): Int?

    // Reload media detail fragment once media is nominated
    fun refreshNominatedMedia(index: Int)
}
